embedded/tbtree/reader.go

```diff
@@ -262,3 +262,14 @@ func (r *Reader) Close() error {
 
 	return nil
 }
+
+func cp(s []byte) []byte {
```

Comment:
If a coy is needed , do we modify keys in-place then?

Reply:
returning a copy is to prevent any external modification (conservative but safe approach).
e.g. sql layer was mapping keys and causing accidental changes, while it'd be possible to make a copy before the mapping, returning a copy will ensure this situation won't take place.